src/logging.js

```diff
@@ -1,41 +1,49 @@
+// @flow
+
 /**
  * Logging setup for our application.
  */
-const stream = require("stream");
-const expressWinston = require("express-winston");
-const winston = require("winston");
+import stream from "stream";
+import expressWinston from "express-winston";
+import winston from "winston";
+
+import {LoggingWinston as StackdriverTransport} from "@google-cloud/logging-winston";
 
-const StackdriverTransport = require("@google-cloud/logging-winston")
-    .LoggingWinston;
-const args = require("./arguments.js");
+import type {NpmLogLevels, Format} from "winston";
+import args from "./arguments.js";
+
+import type {Info, Logger, LogLevel} from "./types.js";
 
 /**
  * This is how the log message gets formatted.
  * We're adding the durationMs from the profiler commands in winston so
  * that we can read them :)
  */
-const prodFormatter = ({message, durationMs}) =>
+const prodFormatter = ({message, durationMs}: Info): string =>
     `${message} ${(durationMs && `(${durationMs}ms)`) || ""}`;
 
 /**
  * Our dev format adds the log level.
  */
-const devFormatter = (info) => `${info.level}: ${prodFormatter(info)}`;
+const devFormatter = (info: Info): string =>
+    `${info.level}: ${prodFormatter(info)}`;
 
 function getFormatters(json, isDev) {
-    const formatters = [
+    const formatters: Array<Format> = [
         winston.format.splat(), // Allows for %s style substitutions
     ];
 
     if (json) {
         formatters.push(winston.format.json());
         if (isDev) {
-            formatters.push(winston.format.prettyPrint({colorize: true}));
+            formatters.push(winston.format.prettyPrint());
         }
     } else {
         formatters.push(winston.format.cli({all: isDev}));
         formatters.push(
-            winston.format.printf(isDev ? devFormatter : prodFormatter),
+            winston.format.printf((info: any) =>
```

Comment:
Huh - what effect did this change have? It doesn't seem like it'd be that much since it's typed as `any`?

Reply:
I wanted the code to be explicit about what we wanted it to be when reading this line rather than tracing through to the receiver of the call. I thought it helped make it easier to grok what was happening.

Reply:
Oh, and the `info` type is not typed properly to be covariant; so the type being passed in causes a flow error when it is passed to our handlers.